### ISSUE TITLE
Interpolate should bail early if an exact distance match is found

### ISSUE THREAD

**Issue by KilianB** [Turfjs/turf#2743]
Is there a logic error in the fast path of the following code? If the certeroid in interpolate is an exact match `zw` is set to the value of the feature but the forEach loop is not aborted. 

This means that we should have a different result depending on which order the points are checked. If the point is found at the end of the feature set the zValue will be taken from the feature, if it's one of the first points it would just be accounted for as normal.

https://github.com/Turfjs/turf/blob/1dec2f2474ba37ca3cb40e492ba85f70aecd055a/packages/turf-interpolate/index.js#L94C1-L95C1

**Comment by smallsaucepan** [comment]
Thanks for raising this @KilianB. Would you mind taking a look at #1601 and seeing how it's related? Are the two issues suggesting the same outcome for different reasons?

**Comment by lemonig** [comment]
> Thanks for raising this [@KilianB](https://github.com/KilianB). Would you mind taking a look at [#1601](https://github.com/Turfjs/turf/issues/1601) and seeing how it's related? Are the two issues suggesting the same outcome for different reasons?

Yeah, Even though the grid points rarely match up exactly with the known points (`d === 0`), 
Nevertheless, the sequence of values is no longer pertinent.because ` zw / sw` will be `Infinity /Infinity `
it's still an issue and we should break out of the loop.  when `featureEach ` function `return false` will be broken loop
